### PR TITLE
Disable conditional allow overlap

### PR DESCRIPTION
**Pull request by IagoLast**
[#58] As discussed in #55 and documented in [the docs](https://github.com/CartoDB/tangram-carto/blob/master/doc/about_rendering.md#dynamic-marker-allow-overlap-is-not-supported) once this PR is merged an error will be thrown if conditional `marker-allow-overlap` is found.

Aceptance criteria
---

- [x] Load some css where the `marker-allow-overlap` is not conditional and check there is no errors.
- [x] Load some css where the `marker-allow-overlap` is conditional and check an error is thrown.

### CODE REVIEW

File: test/unit/points.js [hunk 1]
```diff
@@ -8,15 +8,15 @@ import { getCollide } from '../../src/basic/points';
 
 describe('Points', () => {
   describe('.getCollide', () => {
-    xit('Should return false when marker-allow-overlap is not defined', () => {
+    it('Should return false when marker-allow-overlap is not defined', () => {
       const c3ss = Utils.getShader('#layer {  marker-line-color: #FFF; }');
       expect(getCollide(c3ss)).to.equal(false);
```
Comment (IagoLast) [inline test/unit/points.js]: The reason for the `xit` is that this is returning "false" (string) instead `false`(boolean)

File: test/unit/points.js [hunk 2]
```diff
@@ -8,15 +8,15 @@ import { getCollide } from '../../src/basic/points';
...
     });
     it('Should return false when marker-allow-overlap is true', () => {
       const c3ss = Utils.getShader('#layer { marker-allow-overlap: true; }');
       expect(getCollide(c3ss)).to.equal(true);
     });
-    it('Should return false when marker-allow-overlap is false', () => {
+    it('Should return true when marker-allow-overlap is false', () => {
```
Comment (IagoLast) [inline test/unit/points.js]: The `it` message doesn't match the assertion!

File: src/basic/points.js [hunk 3]
```diff
@@ -78,10 +78,20 @@ const getWidths = compose(
...
+  let property = c3ss['marker-allow-overlap'];
+  if (!property) {
+    return PR['allow-overlap']['default-value'];
+  }
+  // We dont support filtered marker-allow-overlap
+  if (property.filtered) {
+    throw new Error('marker-allow-overlap is not supported inside filters');
+  }
+  // Since this property is not-dynamic must be evaluated.
+  return property.style({}, { zoom: 10 });
```
Comment (IagoLast) [inline src/basic/points.js]: totally agree, but this implied changing the unit-tests too, so I let them to make the review easier.

We can change it now 👍

File: test/unit/points.js (hunk 2 above)
Comment (alonsogarciapablo) [inline test/unit/points.js]: I changed the wrong test case... Fixed again. Good catch!

File: src/basic/points.js (hunk 3 above)
Comment (IagoLast) [inline src/basic/points.js]: ok, good to know I will pass empty objects

File: test/unit/points.js (hunk 1 above)
Comment (alonsogarciapablo) [inline test/unit/points.js]: Test seems to be passing with this change... Isn't this similar to other test cases for .getCollide?

**Comment by IagoLast** [comment]
Regarding to the zoom issue.

The [zoom is not in the filters array](https://github.com/CartoDB/carto/blob/master/lib/carto/tree/definition.js#L214) (we have `this.filters` and `this.zoom`).

Carto generates the `js` rules adding a "zoom" filter by default so when there are no filters the rules are like this:

```javascript
if(zoom < infinity) {  // Always true
  _value = 'foo';
}
```

This was making the `filtered` value to always return true.

I changed that and I removed a bug to create a new one!

I guess I have to change the CARTO code again, to consider the zoom for the `filtered` field only when the zoom is smaller than the maximun value.

**Comment by alonsogarciapablo** [comment]
> I guess I have to change the CARTO code again, to consider the zoom for the filtered field only when the zoom is smaller than the maximum value.

I opened an issue to take care of this: https://github.com/CartoDB/carto/issues/40

**Pull request merged**